### DISCUSSION TITLE
Comparation of FastSpeech2_HifiGan_E2E and FastSpeech2+HiFiGAN

### DISCUSSION THREAD

**Discussion by humanlost** [NVIDIA-NeMo/NeMo#2587] [Q&A] [answered]
Hi,
I download the models of FastSpeech2_HifiGan_E2E and FastSpeech2+HiFiGAN, and generate audios by them.
I find out that the voice quality of FastSpeech2+HiFiGAN is better than FastSpeech2_HifiGan_E2E.
Does this results match your expectations? 
In fact, I I'm going to implement E2E TTS structure recentlty. So I hope to discuss the feasibility of this framework.
Thank you~

**Comment by blisc** [comment, marked as answer]
From our experiments on the models. The two-stage process of spectrogram generator + vocoder gives higher audio quality than comparable end-to-end models.
We have included the end-to-end models for researchers to start with and experiment more on them to see if we can create an end-to-end model that is faster, easier to use, and gives higher quality than the two stage setup.

**Reply by humanlost** [reply]
Thanks, this is very helpful to me~
